projects/laji/src/app/shared/service/geo-convert.service.ts

```diff
@@ -92,17 +102,20 @@ export class GeoConvertService {
     return this.httpClient.post<string>('/api/geo-convert/' + fileId, data, {params});
   }
 
-  private getOutputLink(conversionId: string) {
+  private getResponse(conversionId: string): Observable<GeoConversionResponse> {
     return interval(this.pollInterval).pipe(
-      mergeMap(() => this.getGeoConversionStatus(conversionId)),
-      first(result => result.status === 'complete'),
-      map(() => '/api/geo-convert/output/' + conversionId)
+      concatMap(() => this.getGeoConversionStatus(conversionId)),
+      takeWhile(result => result.status !== 'complete', true),
+      map((result) => ({
+        status: result.status,
+        progressPercent: result.progress_percent,
+        outputLink: result.status === 'complete' ? '/api/geo-convert/output/' + conversionId : null
```

Comment:
Same reasoning for `undefined` vs `null`, `undefined` would be better if there isn't some specific reason for using `null`. The compiler will show an error for the typing of `outputLink` for `null` value once we enable `strictNullChecks` for TS.